standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/IMetaStoreClient.java

```diff
@@ -3478,6 +3478,17 @@ CompactionResponse compact2(String dbname, String tableName, String partitionNam
    */
   ShowCompactResponse showCompactions() throws TException;
 
+  /**
+   * Submit a request for performing cleanup of output directory. This is particularly
+   * useful for CTAS when the query fails after write and before creation of table.
+   * @return Status of whether the request was successfully submitted. True indicates
+   * the request was successfully submitted and false indicates failure of request submitted.
+   * @throws TException
+   */
+  boolean submitForCleanup(String dbname, String tableName, CompactionType type,
```

Comment:
please use a request object, so it could be modified in future

Reply:
As discussed offline, this was already done. The request is created inside this function which actually makes the request to the HMS.

Reply:
Done